hatch/src/hatch_error/mod.rs

```diff
@@ -0,0 +1,81 @@
+use std::fmt;
+use std::error::Error;
+
+use std::io;
+use yaml_rust::scanner;
+
+pub mod null;
+pub mod empty_config;
+pub mod missing_build;
+pub mod missing_name;
+pub mod missing_version;
+
+
+pub use self::null::NullError;
+pub use self::empty_config::EmptyConfigError;
+pub use self::missing_name::MissingNameError;
+pub use self::missing_build::MissingBuildError;
+pub use self::missing_version::MissingVersionError;
+
+#[derive(Debug)]
+pub enum HatchError {
+  Io(io::Error),
+  Parsing(scanner::ScanError),
+  Null(NullError),
+  EmptyConfig(EmptyConfigError),
+  MissingName(MissingNameError),
+  MissingBuild(MissingBuildError),
+  MissingVersion(MissingVersionError),
+}
+
+impl Error for HatchError {
+  fn description(&self) -> &str {
+    match *self {
+      HatchError::Io(ref e) => e.description(),
+      HatchError::Parsing(ref e) => e.description(),
+      HatchError::Null(ref e) => e.description(),
+      HatchError::EmptyConfig(ref e) => e.description(),
+      HatchError::MissingName(ref e) => e.description(),
+      HatchError::MissingBuild(ref e) => e.description(),
+      HatchError::MissingVersion(ref e) => e.description(),
+    }
+  }
+
+  fn cause(&self) -> Option<&Error> {
+    match *self {
+      HatchError::Io(ref e) => e.cause(),
+      HatchError::Parsing(ref e) => e.cause(),
+      HatchError::Null(ref e) => e.cause(),
+      HatchError::EmptyConfig(ref e) => e.cause(),
+      HatchError::MissingName(ref e) => e.cause(),
+      HatchError::MissingBuild(ref e) => e.cause(),
+      HatchError::MissingVersion(ref e) => e.cause(),
+    }
+  }
+}
+
+impl fmt::Display for HatchError {
+  fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+    match *self {
+      HatchError::Io(ref e) => write!(f, "IO error: {}", e),
+      HatchError::Parsing(ref e) => write!(f, "Parsing error: {}", e),
+      HatchError::Null(ref e) => write!(f, "{}", e),
+      HatchError::EmptyConfig(ref e) => write!(f, "Config error: {}", e),
+      HatchError::MissingName(ref e) => write!(f, "Config error: {}", e),
+      HatchError::MissingBuild(ref e) => write!(f, "Config error: {}", e),
+      HatchError::MissingVersion(ref e) => write!(f, "Config error: {}", e),
+    }
+  }
+}
```

Comment:
There are three methods in this file with match expressions matching the type of error. In each case, we call the same method, e.g. `e.description()` or `write!(f, "message", e)`. **Can this be simpler?** I don't expect these errors to change much, but if we add new errors we have to make sure we add cases in all the match expressions.

Reply:
All of the errors _statically_ dispatch to the correct call so the size must be known at compile time.  Ive had this same idea and will look into possibly using generics to make this simpler.

Reply:
👌